mmcv/runner/hooks/logger/pavi.py

```diff
@@ -147,3 +171,20 @@ def before_epoch(self, runner) -> None:
             image = data[self.img_key][0:1].to(device)
             with torch.no_grad():
                 self.writer.add_graph(_model, image)
+
+    @master_only
+    def after_train_epoch(self, runner) -> None:
+        # Do not use runner.epoch since it starts from 0.
+        step = self.get_epoch(runner) if self.by_epoch else self.get_iter(
```

Comment:
If the user is using `IterBasedRunner`, `_add_ckpt` should be written in `after_train_iter` https://github.com/open-mmlab/mmcv/blob/d409eedc816fccfb1c8d57e5eed5f03bd075f327/mmcv/runner/iter_based_runner.py#L70